libs/entry-components/configure-theming.md

```diff
@@ -0,0 +1,435 @@
+    size: 20px
+   ),
+   buttons: (
+    family: 'Roboto',
+    size: 10px
+   )
+  ),
+ )
+);
+```
+
+To apply it on whole body of the app, add `.mat-body` class to the body element in index.html file in app root
+
+```html
+<!-- Applying mat-body class to the root component in index.html -->
+<body class="mat-body">
+<app-root></app-root>
+</body>
+```
+
+Since we're overriding Angular Material, it is important to add CSS classes for mixin that emits styles for native header elements scoped within the .mat-typography CSS class
+
+| CSS class      | Example                                       |
+| -------------- | --------------------------------------------- |
+| .mat-headline-1   | `<h1 class="mat-headline-1">{{title}}</h1>`|
+| .mat-headline-2   | `<h2 class="mat-headline-2">{{title}}</h2>`|
+
+#### Spacing
+
+The submap providing the necessary spacing information for styling the **entry form and search form** **button** and **form field**
+
+```scss
+$custom-theme: (
+ general: (
+  spacing: (
+   default: 15px
+  ),
+ )
+);
+```
+
+#### Buttons
+
+Additional customization  for buttons:
+
+- `icon-size` - defines the default icon size for buttons.
+
+Code Example of general submap configuration:
+
+```scss
+$custom-theme: (
+ general: (
+  buttons: (
+   icon-size: 48px
+  )
+ )
+)
+```
+
+### Tables Configuration
+
+The `tables` section in the `$custom-theme` configuration handles how table components are styled and themed.
+
+#### Cells
+
+Contains properties related to individual cells in a table:
+
+- `edge-gap` specifies the gap between cell edges.
+- `padding` allows customization of cell padding.
+
+#### Rows
+
+Handles the appearance of table rows:
+
+- `selected-color` sets the background color for selected rows.
+- `disabled-color` defines the background color for disabled rows.
+- `odd-even-row` determines whether odd or even rows are styled differently.
+- `odd-even-background` sets the background color for odd or even rows.
+
+#### Contents
+
+Manages the styling of table content:
+
+- `no-result` contains properties for the appearance of a message displayed when there are no search results:
+- `font-size` specifies the font size of the message.
+- `font-weight` controls the font weight of the message.
+
+```scss
+$custom-theme: (
+ tables: (
+  cells: (
+   edge-gap: 4px,
+   padding: null
+  ),
+  rows: (
+   selected-color: #FFF,
+   disabled-color: #F5F5F5,
+   odd-even-row: odd,
+   odd-even-background: #F0F0F0
+  ),
+  contents: (
+   no-result: (
+    font-size: 13px,
+    font-weight: 500
+   )
+  )
+ )
+);
+```
+
+### Dialogs Configuration
+
+The `dialogs` submap within `$custom-theme` focuses on configuring the appearance of dialog components.
+
+#### Title
+
+Includes properties related to the title of a dialog:
+
+- `size` determines the font size of the dialog title.
+
+```scss
+$custom-theme: (
+ dialogs: (
+  title: (
+  size: 20px
+ )
+)
+);
+```
+
+## 2. Native Angular Material Configuration
+
```

Comment:
We should also say that this approach must be used when project has a need for a lot of illogical angular material overrides that we don't wanna support for regular theming and in cases when we want to have more granular control over variants of primary and accent colors together with their shadows.